ext/mysql2/result.c

```diff
@@ -709,7 +715,10 @@ static VALUE rb_mysql_result_fetch_row(VALUE self, MYSQL_FIELD * fields, const r
         case MYSQL_TYPE_STRING:     /* CHAR or BINARY field */
         case MYSQL_TYPE_SET:        /* SET field */
         case MYSQL_TYPE_ENUM:       /* ENUM field */
-        case MYSQL_TYPE_GEOMETRY:   /* Spatial fielda */
+        case MYSQL_TYPE_GEOMETRY:   /* Spatial field */
+#ifdef HAVE_MYSQL_TYPE_JSON
+        case MYSQL_TYPE_JSON:       /* JSON field */
+#endif
         default:
```

Comment:
If the type is unknown, it will fall through to the default, so this isn't strictly necessary